src/common/media.cpp

```diff
@@ -272,6 +275,13 @@ void media_loop(void) {
                     }
                     break;
                 }
+                while (len == MAX_CMD_SIZE && buffer[MAX_CMD_SIZE - 1] != '\n') {
+                    // Read until the end of the line
+                    if (!f_gets(buffer, sizeof(buffer), &media_print_fil)) {
+                        goto no_data;
```

Comment:
I know this is already an if-hell suitable for refactor. However, I would avoid `goto`. You can create `media_no_data()`. Then you'd place here:
```
media_no_data();
return;
```
and this instead of `no_data:` block:
```
} else {
    media_no_data();
    return;
}
```